core/src/mast/serialization/info.rs

```diff
@@ -44,7 +44,8 @@ impl MastNodeInfo {
 
     pub fn try_into_mast_node(
         self,
-        mast_forest: &mut MastForest,
+        mast_forest: &MastForest,
+        node_count: usize,
         basic_block_data_decoder: &BasicBlockDataDecoder,
     ) -> Result<MastNode, DeserializationError> {
```

Comment:
I'd probably add a brief doc-comment to this method. For example, to explain that `node_count` is the expected node count of the `mast_forest` post deserialization.